src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureTests.Dependencies.cs

```diff
@@ -68,9 +68,9 @@ public void ProgramMethod()
     }
 }";
             var test = SetupDependencyAndTestCSWithOneSourceFile(csCurrentAssemblyCode, csDependencyCode);
-            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("AMethod"));
-            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(1).WithArguments("Library"));
-            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(2).WithArguments("AProperty"));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(0).WithArguments("AMethod", DetectPreviewFeatureAnalyzer.DefaultURL));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(1).WithArguments("Library", DetectPreviewFeatureAnalyzer.DefaultURL));
+            test.ExpectedDiagnostics.Add(VerifyCS.Diagnostic(DetectPreviewFeatureAnalyzer.GeneralPreviewFeatureAttributeRule).WithLocation(2).WithArguments("AProperty", DetectPreviewFeatureAnalyzer.DefaultURL));
```

Comment:
Why are we mising diagnostic styles here?

either we don't need the `{||}` parse holes in `csCurrentAssemblyCode` or we don't need to manually specify the diagnostics here right?

Reply:
> Why are we mising diagnostic styles here?

What do diagnostic styles mean? I don't know that term :/

Hmm, not sure I completely understand, but the analyzer reports multiple diagnostics, so I need to use the `{| |}` syntax and add the diagnostics explicitly in the `ExpectedDiagnostics.Add` method right? Is there another way?

Reply:
> but the analyzer reports multiple diagnostics

But the additional diagnostics have locations right? So you can do `{|{||}|}` to nest diagnostics whose locations overlap. I would prefer we use the brackets syntax as its easier to visually understand where the diagnostics are appearing.

Reply:
> But the additional diagnostics have locations right?

Wait maybe I wasn't being clear. I meant that DetectPreviewFeaturesAnalyzer can report ~17 different diagnostics, not that it's reporting the same diagnostic in multiple places. So the only option in the unit tests is to use **both** the `{| |}` syntax **and** the `ExpectedDiagnostics.Add` method IIRC

Reply:
> So you can do {|{||}|} to nest diagnostics whose locations overlap.

@jmarolf  I might be missing something, but i don't see such overlaps in the diagnostics